modules/objdetect/include/opencv2/objdetect.hpp

```diff
@@ -763,10 +763,47 @@ class CV_EXPORTS_W QRCodeEncoder {
 
 };
 
+enum QRDetectMethod{
+    BASE, // old method
+    ARUCO // new method
+    // could add WECHAT method
+};
+
+struct CV_EXPORTS_W_SIMPLE QRDetectorParameters {
+    QRDetectorParameters() {}
+
+    CV_PROP_RW QRDetectMethod qrDetectMethod = QRDetectMethod::ARUCO;
+
+    // The minimum allowed size of a QR module in the smallest image in the image pyramid.
+    CV_PROP_RW float minModuleSizeInPyramid = 4.f;
+
+    // The maximum allowed rotation relative diff for finder patterns in one QR code
+    CV_PROP_RW float maxRotateDiff = (float)CV_PI/12.f;
+
+    // The maximum allowed module relative diff for finder patterns in one QR code
+    CV_PROP_RW float maxRelativeModuleDiff = 1.75f;
+
+    // The maximum allowed module relative diff for timing pattern module
+    // If relative diff of timing pattern module more this values, calculateTimingPattern() will add penalty points.
+    // If calculateTimingPattern() add many penalty points, QR code will be rejected.
+    CV_PROP_RW float maxTimingPatternDiff = 2.f;
+
+    // The maximum allowed percentage of penalty points out of total pins in timing pattern.
+    CV_PROP_RW float maxPenalties = 0.4f;
+
+    // The maximum allowed color difference in the timing pattern.
+    CV_PROP_RW float maxColorsDelta = 0.2f;
+
+    // The algorithm find QR codes with almost minimum timing pattern score and minimum size
+    // For best QR code timingPatternScore*maxTimingScoreDiff < anyQrTimingPatternScore and size < anyQrSize
+    CV_PROP_RW float maxTimingScoreDiff = 0.9f;
+};
```

Comment:
Please format description in Doxygen friendly format to get them in documentation.

Reply:
added @brief